File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        res = 0
        left = 0
        count = defaultdict(int)
        for right in range (len(nums)):
            count[nums[right]] += 1
            while count[nums[right]] > k:
                count[nums[left]] -= 1
                left += 1
            res = max(res, (right -left +1))
        return res
        

File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.java
class Solution {
    public int maxSubarrayLength(int[] nums, int k) {
        int result = 0;
        int left = 0;
        Map<Integer,Integer> freq = new HashMap<>();
        for(int right=0;right<nums.length;right++){
            freq.put(nums[right],freq.getOrDefault(nums[right],0)+1);
            while(freq.get(nums[right])>k){
                freq.put(nums[left],freq.get(nums[left])-1);
                left++;
            }
            result = Math.max(result,(right - left +1));
        }            
        return result;
    }
}